### PR TITLE
Handle time zone environment variable having international characters.

### PR DESCRIPTION
**Pull request by jonsimantov**
Avoid sending the backend the wrong time zone format (or one with international characters) by using the Windows ICU library to get the canonical IANA time zone names (e.g. "America/Los Angeles").

Previously, on Windows, it used the "TZ" environment variable verbatim. This could have contained international characters (see #869). Now, use _tzset() to get the canonical Windows name, and then use the ICU library to convert it to the IANA name, which the Remote Config backend expects.

***

Tested with "Romance Standard Time" --> "Europe/Paris" with LogInfo, prior to changing back to LogDebug.
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Comment by jonsimantov** [comment]
FYI not planning to merge until after the Unity release, just to be safe.

**Comment by jonsimantov** [comment]
Unity build and tests succeeded here: https://github.com/firebase/firebase-unity-sdk/actions/runs/5138069241

File: app/src/locale.cc [hunk 1]
```diff
@@ -53,6 +58,7 @@ std::string GetLocale() {
   std::string output(woutput.begin(), woutput.end());
   // Change all hyphens to underscores to normalize the locale.
   std::replace(output.begin(), output.end(), '-', '_');
+  LogDebug("Locale: %s", output.c_str());
```
Comment (a-maurice) [inline app/src/locale.cc:61]: Still want?

File: app/src/locale.cc [hunk 2]
```diff
@@ -90,22 +96,92 @@ std::string GetLocale() {
...
+  // Get the standard time zone name.
+  std::string windows_tz_utf8;
+  {
+    size_t length = 0;  // get the needed string length
+    if (_get_tzname(&length, nullptr, 0, 0) != 0) return "";
+    std::vector<char> namebuf(length);
+    if (_get_tzname(&length, &namebuf[0], length, 0) != 0) return "";
+    windows_tz_utf8 = std::string(&namebuf[0]);
+  }
+  LogDebug("Windows time zone: %s", windows_tz_utf8.c_str());
```
Comment (a-maurice) [inline app/src/locale.cc:114]: Same here?  If so, it might want a bit of context at least

File: app/src/locale.cc [hunk 3]
```diff
@@ -90,22 +96,92 @@ std::string GetLocale() {
...
+        return windows_tz_utf8;
+      windows_tz_utf8 = std::string(&namebuf[0]);
+    }
+    return windows_tz_utf8;
+  }
+
+  std::wstring iana_tz_utf16(iana_time_zone_buffer);
+  std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> to_utf8;
+  std::string iana_tz_utf8 = to_utf8.to_bytes(iana_tz_utf16);
+  LogDebug("IANA time zone: %s", iana_tz_utf8.c_str());
```
Comment (a-maurice) [inline app/src/locale.cc:182]: And one more

Comment (jonsimantov) [reply app/src/locale.cc:182]: Yeah, I switched them from LogInfo (while I was developing it) to LogDebug but I'll just remove them.

File: app/tests/CMakeLists.txt [hunk 4]
```diff
@@ -207,11 +207,17 @@ firebase_cpp_cc_test(firebase_app_path_test
     firebase_app
 )
 
+if (MSVC)
```
Comment (a-maurice) [inline app/tests/CMakeLists.txt:210]: Can you add a comment for this

**Review by a-maurice — APPROVED** [review]

**Comment by jonsimantov** [comment]
Removed the extraneous log statements and added a comment to the cmake file.

**Review by a-maurice — APPROVED** [review]

**Pull request merged**